yarn/src/main/scala/org/apache/spark/deploy/yarn/security/AMDelegationTokenRenewer.scala

```diff
@@ -171,10 +175,13 @@ private[yarn] class AMDelegationTokenRenewer(
     keytabLoggedInUGI.doAs(new PrivilegedExceptionAction[Void] {
       // Get a copy of the credentials
       override def run(): Void = {
-        val nns = YarnSparkHadoopUtil.get.getNameNodesToAccess(sparkConf) + dst
-        hadoopUtil.obtainTokensForNamenodes(nns, freshHadoopConf, tempCreds)
-        hadoopUtil.obtainTokenForHiveMetastore(sparkConf, freshHadoopConf, tempCreds)
-        hadoopUtil.obtainTokenForHBase(sparkConf, freshHadoopConf, tempCreds)
+        var nearestNextTime = Long.MaxValue
+        credentialManager.obtainCredentials(freshHadoopConf, tempCreds).foreach { nextTime =>
```

Comment:
I also noticed that the logic will renew all tokens when the first token needs renewal. It's probably fine since this should happen very rarely (once every few days). But that also means that `obtainCredentialsFromService` is completely unused and can be deleted.